candi/bashible/bashbooster/50_apt_rpm.sh

```diff
@@ -0,0 +1,194 @@
+    do
+        local NEED_FIRE=false
+        if test -f "$BB_APT_UNHANDLED_PACKAGES_STORE" && grep -Eq "^${PACKAGE}$" "$BB_APT_UNHANDLED_PACKAGES_STORE"; then
+            NEED_FIRE=true
+        fi
+
+        if [[ "$FORCE" == "true" ]] || ! bb-apt-package? "$PACKAGE"; then
+            PACKAGES_TO_INSTALL+=("$PACKAGE")
+        fi
+    done
+
+    if [ "${#PACKAGES_TO_INSTALL[@]}" -gt "0" ]
+    then
+        bb-apt-rpm-update
+        bb-log-info "Installing packages '${PACKAGES_TO_INSTALL[@]}'"
+        apt-get install -y ${PACKAGES_TO_INSTALL[@]}
+        bb-apt-rpm-hold ${PACKAGES_TO_INSTALL[@]}
+        bb-exit-on-error "Failed to install packages '${PACKAGES_TO_INSTALL[@]}'"
+        printf '%s\n' "${PACKAGES_TO_INSTALL[@]}" >> "$BB_APT_UNHANDLED_PACKAGES_STORE"
+        NEED_FIRE=true
+    fi
+    if [[ "$NEED_FIRE" == "true" ]]; then
+        bb-event-fire "bb-package-installed" "$PACKAGE"
+    fi
+}
+
+bb-apt-rpm-remove() {
+    export DEBIAN_FRONTEND=noninteractive
+    local PACKAGES_TO_REMOVE=( )
+
+    for PACKAGE in "$@"
+    do
+        if bb-apt-package? "$PACKAGE"
+        then
+            PACKAGES_TO_REMOVE+=( "$PACKAGE" )
+        fi
+    done
+
+    if [ "${#PACKAGES_TO_REMOVE[@]}" -gt 0 ]; then
+        bb-log-info "Removing packages '${PACKAGES_TO_REMOVE[@]}'"
+        apt-get remove -y ${PACKAGES_TO_REMOVE[@]}
+        bb-exit-on-error "Failed to remove packages '${PACKAGES_TO_REMOVE[@]}'"
+        for i in ${PACKAGES_TO_REMOVE[@]}; do
+            bb-event-fire "bb-package-removed" "$i"
+        done
+    fi
+}
+
+bb-apt-rpm-autoremove() {
+    export DEBIAN_FRONTEND=noninteractive
+    bb-log-info 'Autoremoving unused packages'
+    apt-get --purge -y autoremove
+}
+
+bb-apt-rpm-hold?() {
+    exit 0 #TODO
+    dpkg -s "$1" 2> /dev/null | grep -q '^Status:.\+installed'
+}
+
+bb-apt-rpm-hold() {
+    exit 0 #TODO
+    for PACKAGE in "$@"
+    do
+        apt-mark hold "$PACKAGE"
+    done
+}
+
+bb-apt-rpm-unhold() {
+    for PACKAGE in "$@"
+    do
+        apt-mark unhold "$PACKAGE"
+    done
+}
+
+bb-apt-rpm-package-upgrade?() {
+    bb-apt-rpm-update
+
+    local PACKAGE=$1
+    local OUTPUT="$(
+        apt-cache policy "$PACKAGE" | awk -c '
+            /Installed: / { installed = $2 }
+            /Candidate: / {
+                if (installed != "(none)" && installed != $2) {
+                    print installed " " $2
+                }
+            }
+        '
+    )"
+
+    # Note: No upgrade available is reported for a non-installed package
+    [[ -n "$OUTPUT" ]]
+}
+
+bb-apt-rpm-upgrade() {
+    for PACKAGE in "$@"
+    do
+        if bb-apt-rpm-package-upgrade? "$PACKAGE"
+        then
+            bb-log-info "Upgrading package '$PACKAGE'"
+            bb-event-fire "bb-package-pre-upgrade" "$PACKAGE"
+            apt-get upgrade -y "$PACKAGE"
+            bb-exit-on-error "Failed to upgrade package '$PACKAGE'"
+            bb-event-fire "bb-package-post-upgrade" "$PACKAGE"
+        fi
+    done
+}
```

Comment:
I see this functions is very similar to https://github.com/deckhouse/deckhouse/blob/main/candi/bashible/bashbooster/50_apt.sh
Why we need completely different functions for Alt ?

Reply:
Altlinux does not have apt and apt-get does not have debian-specific flags.